hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/type/SqlLazyDeserializationTest.java

```diff
@@ -0,0 +1,151 @@
+
+import com.hazelcast.config.Config;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.internal.serialization.Data;
+import com.hazelcast.internal.serialization.InternalSerializationService;
+import com.hazelcast.internal.serialization.impl.InternalGenericRecord;
+import com.hazelcast.jet.impl.util.Util;
+import com.hazelcast.nio.serialization.ClassDefinition;
+import com.hazelcast.nio.serialization.ClassDefinitionBuilder;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.experimental.categories.Category;
+
+import java.io.IOException;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import static com.hazelcast.jet.sql.SqlTestSupport.assertRowsAnyOrder;
+import static com.hazelcast.jet.sql.SqlTestSupport.setupCompactTypesForNestedQuery;
+import static com.hazelcast.jet.sql.SqlTestSupport.rows;
+import static com.hazelcast.jet.sql.SqlTestSupport.setupPortableTypesForNestedQuery;
+import static com.hazelcast.spi.properties.ClusterProperty.SQL_CUSTOM_TYPES_ENABLED;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static com.hazelcast.jet.sql.impl.type.SqlLazyDeserializationTestUtil.ResultCaptor;
+import static com.hazelcast.jet.sql.impl.type.SqlLazyDeserializationTestUtil.SqlLazyDeserializationTestInstanceFactory;
+
+/**
+ * This test tests if nested field access for compact and portable uses lazy deserialization or not.
+ * This test does not extend SqlTestSupport on purpose because we want full control over the member
+ * we start.
+ */
+@RunWith(HazelcastSerialClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class SqlLazyDeserializationTest {
+
+    private final SqlLazyDeserializationTestInstanceFactory mockInstanceFactory
+            = new SqlLazyDeserializationTestInstanceFactory();
+
+    private HazelcastInstance instance;
+    private HazelcastInstance client;
+
+    private InternalSerializationService serializationService;
+
+    @Before
+    public void before() {
+        final Config config = new Config();
+        config.getJetConfig().setEnabled(true);
+        config.setProperty(SQL_CUSTOM_TYPES_ENABLED.getName(), "true");
+
+        instance = mockInstanceFactory.newHazelcastInstance(config);
+        serializationService = Util.getHazelcastInstanceImpl(instance).getSerializationService();
+
+        ClassDefinition officeType = new ClassDefinitionBuilder(1, 3)
+                .addLongField("id")
+                .addStringField("name")
+                .build();
+        serializationService.getPortableContext().registerClassDefinition(officeType);
```

Comment:
why don't we configure these class definitions before the instance starts, through the serialization config?